comp/core/healthprobe/healthprobeimpl/healthprobe.go

```diff
@@ -0,0 +1,178 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package healthprobeimpl implements the healthprobe component interface
+package healthprobeimpl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"runtime"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	healthprobeComponent "github.com/DataDog/datadog-agent/comp/core/healthprobe"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/gorilla/mux"
+)
+
+const defaultTimeout = time.Second
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newHealthProbe),
+	)
+}
+
+type dependencies struct {
+	fx.In
+
+	Log    log.Component
+	Config config.Component
+}
+
+type healthprobe struct {
+	config   config.Component
+	log      log.Component
+	server   *http.Server
+	listener net.Listener
+}
+
+func (h *healthprobe) start() error {
+	h.log.Debugf("Health check listening on port %d", h.config.GetInt("health_port"))
+
+	go h.server.Serve(h.listener) //nolint:errcheck
+
+	return nil
+}
+
+func (h *healthprobe) stop() error {
+	timeout, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	h.log.Debug("Stopping Health check")
+	return h.server.Shutdown(timeout) //nolint:errcheck
+}
+
+func newHealthProbe(lc fx.Lifecycle, deps dependencies) (healthprobeComponent.Component, error) {
+	healthPort := deps.Config.GetInt("health_port")
```

Comment:
how does it works with port binding when several agent process expose a health probe?

Reply:
That is a great question. 

I thought there was only one `health_probe` configuration. 

I did a bit of digging. I guess what your are referring is to the system probe which seems to have a dedicated `health_port` configuration https://github.com/DataDog/datadog-agent/pull/19297. Were you referring to that case?

Reply:
@clamoriniere I changed the code so we can get the right port for each agent https://github.com/DataDog/datadog-agent/pull/23911/commits/966c0b6a36de7d7492b359cb273b70413538e6cd

As for handling the port binding conflicts the existing implementation doe snot deal with any of that and the PR https://github.com/DataDog/datadog-agent/pull/19297 hinted that the system probe port is 5558 to avoid collisions